docs/content/en/docs/Coding Milestones/PoC2023q1/edge-syncer.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Edge Syncer"
+date: 2023-03-22
+weight: 4
+description: >
+---
+
+{{% pageinfo %}}
+Edge syncer is a new syncer working with mailbox workspace in edge-mc. 
+{{% /pageinfo %}}
+
+![edge-syncer drawio](./images/edge-syncer-overview.png)
+
+#### Registering Edge Syncer on a p-cluster
+
+Edge-syncer can be deployed on p-cluster easily by the following steps.
+1. Create SyncTarget and Location
+  - Mailbox controller creates mailbox workspace automatically. 
+  - Get mailbox workspace id
+2. Use command to register edge-syncer
+   ```console
+   kubectl ws <mb-ws name>
+   kcp workload edge-sync <EM Sync Target name> --image <EM Syncer Image> -o edge-syncer.yaml
+   ```
+3. Deploy edge-syncer on p-cluster.
+
+![edge-syncer boot](./images/edge-syncer-boot.png)
+
+#### Deploy workload objects from edge-mc to p-cluster
+
+To deploy resources to p-clusters, create the following in workload management workspace
+- workload objects
+  - Some objects are denatured if needed.
+  - Other objects are as it is
+- APIExport/API Schema corresponding to CRD
+  - TBD: Conversion from CRD to APIExport/APISchema could be automated by using MutatingAdmissionWebhook on workload management workspace.
+- EdgeSyncConfig (This is going to be replaced with EdgePlacement spec) 
+
+![edge-syncer deploy](./images/edge-syncer-deploy.png)
+
+After this, Edge-mc will put the following in the mailbox workspace.
+- Workload objects (both denatured one and not-denatured one)
+- API Binding (to API Export on the workload management workspace)
+- EdgeSyncConfig CR
+
+**TODO**: This is something we should clarify..e.g. which existing controller(s) in edge-mc will cover, or just create a new controller to handle uncovered one. 
+
+#### EdgeSyncConfig
+- The example is [here](https://gist.github.com/yuji-watanabe-jp/dfa9cdf35b06f612818446c2c3eb7969).
+- The CR here is used from edge syncer. 
+- The CR is placed at mb-ws to define
+  - object selector
+  - need of renaturing
```

Comment:
I was expecting that the need for denaturing and renaturing is not something that needs to be configured.

In the case of C2P, if we install Kyverno with an Operator, then the Operator will install the Kyverno CRDs in the edge cluster and so those CRDs will not need to be downsynced, right?  If we follow this approach then there is no need for configurable denaturing/renaturing, right?

Reply:
> I was expecting that the need for denaturing and renaturing is not something that needs to be configured. There is a fixed plan outlined in https://docs.kcp-edge.io/docs/coding-milestones/poc2023q1/outline/#data-objects .

Yes, in that case, the fixed plan can be encoded as a default configuration of edge syncer, so need to be configured specifically unless required.

> In the case of C2P, if we install Kyverno with an Operator, then the Operator will install the Kyverno CRDs in the edge cluster and so those CRDs will not need to be downsynced, right? If we follow this approach then there is no need for configurable denaturing/renaturing, right?

In this case, the question is how to install operator. 

> Using an Operator to install CRDs on the edge cluster creates a new category of resources: those that are neither built into the edge cluster nor defined by a downsynced CRD. I think we can start with a hack, making the platform config say that those resources are built into the edge clusters. A more accurate way of handling them would be a way of declaring in the workload management workspace that certain resources will eventually be defined in the edge cluster as a consequence of other things.

Yes, this is the case CRD is created in p-cluster side. We will need to think about this situation later. (maybe post Apr. 27)
- Currently, assuming corresponding API Binding is set on mailbox workspace for upsyncing
- In some case, CRD is created in p-cluster, but CR is created/updated in mailbox workspace for downsync. (typical example is OPA/Gatekeeper Constraint CR/CRD). Currently, assuming the same as above.